lib/ocpp/v201/charge_point.cpp

```diff
@@ -3086,6 +3091,29 @@ void ChargePoint::handle_heartbeat_response(CallResult<HeartbeatResponse> call)
     }
 }
 
+void ChargePoint::handle_set_charging_profile_req(Call<SetChargingProfileRequest> call) {
+    EVLOG_debug << "Received SetChargingProfileRequest: " << call.msg << "\nwith messageId: " << call.uniqueId;
+    auto msg = call.msg;
+    SetChargingProfileResponse response;
+    response.status = ChargingProfileStatusEnum::Rejected;
+
+    auto res = this->smart_charging_handler->validate_profile(msg.chargingProfile, msg.evseId);
+    if (res == ProfileValidationResultEnum::Valid) {
+        EVLOG_debug << "Accepting SetChargingProfileRequest";
+        response = this->smart_charging_handler->add_profile(msg.evseId, msg.chargingProfile);
+        this->callbacks.set_charging_profiles_callback();
+    } else {
+        response.statusInfo = StatusInfo();
+        response.statusInfo->reasonCode = conversions::profile_validation_result_to_reason_code(res);
+        response.statusInfo->additionalInfo = conversions::profile_validation_result_to_string(res);
+        EVLOG_debug << "Rejecting SetChargingProfileRequest:\n reasonCode: " << response.statusInfo->reasonCode.get()
```

Comment:
I think it would be helpful to give the original validation failure in the logs as well. It would help pin point the issue faster.

Reply:
That's already printed as `additionalInfo`?